qt/scientific_interfaces/ISISReflectometry/GUI/Experiment/ExperimentView.cpp

```diff
@@ -0,0 +1,636 @@
+    message += messageFor(missingValues);
+
+  for (auto &typeError : typeErrors)
+    message += messageFor(typeError);
+
+  QMessageBox::warning(
+      this, "Failed to load one or more defaults from parameter file", message);
+}
+
+QLineEdit &ExperimentView::stitchOptionsLineEdit() const {
+  return *static_cast<QLineEdit *>(m_stitchEdit);
+}
+
+/** Creates hints for 'Stitch1DMany'
+ * @param hints :: Hints as a map
+ */
+void ExperimentView::createStitchHints(
+    const std::vector<MantidWidgets::Hint> &hints) {
+
+  // We want to add the stitch params box next to the stitch
+  // label, so first find the label's position
+  auto stitchLabelIndex = m_ui.expSettingsGrid->indexOf(m_ui.stitchLabel);
+  int row, col, rowSpan, colSpan;
+  m_ui.expSettingsGrid->getItemPosition(stitchLabelIndex, &row, &col, &rowSpan,
+                                        &colSpan);
+  // Create the new edit box and add it to the right of the label
+  m_stitchEdit = new MantidWidgets::HintingLineEdit(this, hints);
+  m_ui.expSettingsGrid->addWidget(m_stitchEdit, row, col + colSpan, 1, 3);
+}
+
+double ExperimentView::getCRho() const { return m_ui.CRhoEdit->value(); }
+
+void ExperimentView::setCRho(double cRho) { m_ui.CRhoEdit->setValue(cRho); }
+
+double ExperimentView::getCAlpha() const { return m_ui.CAlphaEdit->value(); }
+
+void ExperimentView::setCAlpha(double cAlpha) {
+  m_ui.CAlphaEdit->setValue(cAlpha);
+}
+
+double ExperimentView::getCAp() const { return m_ui.CApEdit->value(); }
+
+void ExperimentView::setCAp(double cAp) { m_ui.CApEdit->setValue(cAp); }
+
+double ExperimentView::getCPp() const { return m_ui.CPpEdit->value(); }
+
+void ExperimentView::setCPp(double cPp) { m_ui.CPpEdit->setValue(cPp); }
+
+std::string ExperimentView::getFloodCorrectionType() const {
+  return getText(*m_ui.floodCorComboBox);
+}
+
+void ExperimentView::setFloodCorrectionType(std::string const &type) {
+  setSelected(*m_ui.floodCorComboBox, type);
+}
+
+std::string ExperimentView::getFloodWorkspace() const {
+  return getText(*m_ui.floodWorkspaceWsSelector);
+}
+
+void ExperimentView::setFloodWorkspace(std::string const &workspace) {
+  setSelected(*m_ui.floodWorkspaceWsSelector, workspace);
+}
+
+std::string ExperimentView::getAnalysisMode() const {
+  return getText(*m_ui.analysisModeComboBox);
+}
+
+void ExperimentView::setAnalysisMode(std::string const &analysisMode) {
+  setSelected(*m_ui.analysisModeComboBox, analysisMode);
+}
+
+std::string ExperimentView::getSummationType() const {
+  return getText(*m_ui.summationTypeComboBox);
+}
+
+void ExperimentView::setSummationType(std::string const &summationType) {
+  return setSelected(*m_ui.summationTypeComboBox, summationType);
+}
+
+std::string ExperimentView::getReductionType() const {
+  return getText(*m_ui.reductionTypeComboBox);
+}
+
+bool ExperimentView::getIncludePartialBins() const {
+  return m_ui.includePartialBinsCheckBox->isChecked();
+}
+
+void ExperimentView::setIncludePartialBins(bool enable) {
```

Comment:
I think this bool can be const'd (same in setDebugOption) -- setChecked doesn't mutate `enable`

Reply:
True, but the local style in this file is to not use const when passing by value. I don't think our coding standards mention this so I tend to go with the local style.